examples/ContinuumSnakeWithLiftingWaveCase/snake_forcing.py

```diff
@@ -0,0 +1,201 @@
+                    Angular frequency of traveling wave.
+            wave_number: float
+                    Wave number of traveling wave.
+            phase_shift: float
+                    Phase shift of traveling wave.
+            ramp_up_time: float
+                    Applied muscle torques are ramped up until ramp up time.
+            my_spline: numpy.ndarray
+                    1D (blocksize) array containing data with 'float' type. Generated spline.
+            switch_on_time: float
+                    time to switch on the muscle activation.
+    """
+
+    def __init__(
+        self,
+        base_length,
+        b_coeff,
+        period,
+        wave_number,
+        phase_shift,
+        direction,
+        rest_lengths,
+        ramp_up_time=0.0,
+        with_spline=False,
+        switch_on_time=0.0,
+        *args,
+        **kwargs,
+    ):
+        """
+
+        Parameters
+        ----------
+        base_length: float
+                Rest length of the rod-like object.
+        b_coeff: nump.ndarray
+                1D array containing data with 'float' type.
+                Beta coefficients for beta-spline.
+        period: float
+                Period of traveling wave.
+        wave_number: float
+                Wave number of traveling wave.
+        phase_shift: float
+                Phase shift of traveling wave.
+        direction: numpy.ndarray
+           1D (dim) array containing data with 'float' type.
+        ramp_up_time: float
+                Applied muscle torques are ramped up until ramp up time.
+        with_spline: boolean
+                Option to use beta-spline.
+        switch_on_time: float
+                time to switch on the muscle activation.
+
+        """
+        super(MuscleTorques_lifting, self).__init__()
+
+        self.direction = direction  # Direction torque applied
+        self.angular_frequency = 2.0 * np.pi / period
+        self.wave_number = wave_number
+        self.phase_shift = phase_shift
+
+        assert ramp_up_time >= 0.0
+        self.ramp_up_time = ramp_up_time
+        assert switch_on_time >= 0.0
+        self.switch_on_time = switch_on_time
+
+        # s is the position of nodes on the rod, we go from node=1 to node=nelem-1, because there is no
+        # torques applied by first and last node on elements. Reason is that we cannot apply torque in an
+        # infinitesimal segment at the beginning and end of rod, because there is no additional element
+        # (at element=-1 or element=n_elem+1) to provide internal torques to cancel out an external
+        # torque. This coupled with the requirement that the sum of all muscle torques has
+        # to be zero results in this condition.
+        self.s = np.cumsum(rest_lengths)
+        self.s /= self.s[-1]
+
+        if with_spline:
+            assert b_coeff.size != 0, "Beta spline coefficient array (t_coeff) is empty"
+            my_spline, ctr_pts, ctr_coeffs = _bspline(b_coeff)
```

Comment:
Naming an instance variable `my_spline` feels weird. I would suggest renaming it to something like `spline_func` and `spline` but it's up to you.